script_runner/frontend/src/Input.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from "react";
+import { ParamType } from "./types";
+
+
+type Props = {
+  id: string;
+  disabled: boolean;
+  value: string;
+  onChange: (value: string) => void;
+  type: ParamType;
+  // applies to autocomplete and dynamic_autocomplete
+  options: string[] | null;
+}
```

Comment:
Any specific reason for not defining this in types.tsx? I see you already have this file. Curious to know if there's some sort of pattern on when to define types in files versus generically in types.tsx

Reply:
the props for each component are always kept in the same file as the component since they are should not be used anywhere else except the function signature of that component, and is not meant to be a generally reusable type.

i think this is a fairly standard pattern. you can see it used in sentry too. example: https://github.com/getsentry/sentry/blob/7f6722f90a0f121b45aea530158a28dc5ac0f7b3/static/app/views/app/systemAlerts.tsx#L6